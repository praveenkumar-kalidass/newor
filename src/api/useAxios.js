import axios from 'axios';

import useLoader from 'provider/Loader/useLoader';
import config from '../../config/config';

const useAxios = () => {
  const { addLoader, removeLoader } = useLoader();

  const instance = axios.create({
    baseURL: config.baseURL,
    timeout: 1000,
  });
  instance.interceptors.request.use((options) => {
    if (options.loader) {
      addLoader();
    }
    return options;
  }, (error) => (
    Promise.reject(error)
  ));
  instance.interceptors.response.use((response) => {
    if (response.config.loader) {
      removeLoader();
    }
    return response;
  }, (error) => {
    if (error.config.loader) {
      removeLoader();
    }
    return Promise.reject(error);
  });

  return instance;
};

export default useAxios;
