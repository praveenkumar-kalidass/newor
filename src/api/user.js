import axios from './index';
import config from '../../config/config';
import CONSTANT from '../constant';

export const login = ({ email, password }) => axios.post(`${config.baseURL}/api/user/v1/login`, {
  email,
  password,
  clientId: config.clientId,
  clientSecret: config.clientSecret,
  grantType: CONSTANT.AUTH_LITERAL.PASSWORD,
  responseType: CONSTANT.AUTH_LITERAL.CODE,
});

export const signup = (request) => axios.post(`${config.baseURL}/api/user/v1/signup`, request);
