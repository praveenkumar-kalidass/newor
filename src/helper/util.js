import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
