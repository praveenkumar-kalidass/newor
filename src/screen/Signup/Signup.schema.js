import Joi from 'joi';

import TRANSLATION from '../../translation/en.json';

export const SCHEMA = Joi.object({
  firstName: Joi.string().alphanum().min(3).max(25)
    .required(),
  lastName: Joi.string().alphanum().min(3).max(25)
    .required(),
  email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com'] } }),
  password: Joi.string().alphanum().min(6).max(15)
    .required(),
});

export const INITIAL_STATE = {
  firstName: '',
  lastName: '',
  email: '',
  password: '',
};

export const FIELDS = [
  {
    key: 'firstName',
    placeholder: TRANSLATION.FIRST_NAME,
  },
  {
    key: 'lastName',
    placeholder: TRANSLATION.LAST_NAME,
  },
  {
    key: 'email',
    placeholder: TRANSLATION.EMAIL,
  },
  {
    key: 'password',
    placeholder: TRANSLATION.PASSWORD,
  },
];
