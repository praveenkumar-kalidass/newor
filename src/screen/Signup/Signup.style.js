import styled from 'styled-components/native';
import {
  Box, Button, Input, Text,
} from 'native-base';

import COLOR from '../../constant/color';

export const FormContainer = styled(Box).attrs(() => ({
  borderRadius: 20,
  flex: 1,
  alignSelf: 'center',
  p: 5,
  bg: COLOR.LIGHT_100,
}))``;

export const InputField = styled(Input).attrs(({ isInvalid }) => ({
  _focus: {
    bg: COLOR.LIGHT_100,
    borderColor: COLOR.LIGHT_SECONDARY_100,
  },
  mb: isInvalid ? 0 : 5,
}))`
  font-size: 14px;
`;

export const SubmitButton = styled(Button).attrs(() => ({
  bg: COLOR.LIGHT_PRIMARY_100,
  _pressed: { bg: COLOR.LIGHT_PRIMARY_20 },
  _text: { fontWeight: 'bold' },
  mb: 5,
}))``;

export const Title = styled(Text).attrs(() => ({
  bold: true,
  fontSize: 20,
}))`
  text-transform: capitalize;
`;

export const ToastBox = styled(Box).attrs(({ isError }) => ({
  _text: { color: COLOR.LIGHT_100 },
  bg: isError ? COLOR.LIGHT_ERROR : COLOR.LIGHT_SUCCESS,
  p: 5,
  rounded: 5,
}))``;

export const LoginLink = styled(Text).attrs(({ link }) => ({
  fontSize: 'md',
  underline: link,
  color: link && COLOR.LIGHT_LINK,
}))``;
