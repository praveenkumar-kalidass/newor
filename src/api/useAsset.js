import useAxios from '.';
import config from '../../config';

const useAsset = () => {
  const axios = useAxios();

  const addDeposit = (request) => axios.post(`${config.baseURL}/api/deposit/v1`, request);

  const getAsset = (id) => axios.get(`${config.baseURL}/api/asset/v1/${id}`);

  return {
    addDeposit,
    getAsset,
  };
};

export default useAsset;
