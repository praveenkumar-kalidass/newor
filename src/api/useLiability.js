import useAxios from '.';
import config from '../../config';

const useLiability = () => {
  const axios = useAxios();

  const getLiability = (id) => axios.get(`${config.baseURL}/api/liability/v1/${id}`);

  const addLoan = (request) => axios.post(`${config.baseURL}/api/loan/v1`, request);

  return {
    getLiability,
    addLoan,
  };
};

export default useLiability;
