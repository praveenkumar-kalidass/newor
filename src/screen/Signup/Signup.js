import React, {
  useCallback, useEffect, useMemo, useState,
} from 'react';
import { useNavigation } from '@react-navigation/native';
import {
  Center, Flex, FormControl, Image, useToast, AlertDialog,
} from 'native-base';

import TitleBar from 'component/TitleBar/TitleBar';
import ToastAlert from 'component/ToastAlert';
import AppButton from 'component/AppButton';
import CONSTANT from 'constant';
import ROUTE from 'constant/route';
import NEWOR from 'asset/image/newor.png';
import NEWOR_SUCCESS from 'asset/image/newor-success.png';
import validator from 'helper/validator';
import { signup } from 'api/user';
import useTheme from 'theme/useTheme';
import withBackground from 'helper/withBackground';
import { Background2 } from 'component/Background';
import useTranslation from 'translation/useTranslation';
import Translation from 'translation/Translation';
import {
  FIELDS,
  INITIAL_STATE,
  SCHEMA,
} from './Signup.schema';
import {
  FormContainer,
  InputField,
  Title,
  LoginLink,
} from './Signup.style';

const Signup = () => {
  const [fields, setFields] = useState(INITIAL_STATE);
  const [errorMessages, setErrorMessages] = useState(INITIAL_STATE);
  const [isSubmit, setIsSubmit] = useState(false);
  const [isSuccess, setIsSuccess] = useState(false);
  const navigation = useNavigation();
  const toast = useToast();
  const theme = useTheme();
  const { translate } = useTranslation();

  const handleFieldChange = (label, value) => {
    setFields({
      ...fields,
      [label]: value,
    });
  };

  const handleValidation = () => {
    setErrorMessages(validator(SCHEMA, fields, INITIAL_STATE));
  };

  const handleSubmit = () => {
    setErrorMessages(validator(SCHEMA, fields, INITIAL_STATE, true));
    setIsSubmit(true);
  };

  const isFormError = useMemo(
    () => Object.keys(errorMessages).some(
      (field) => !fields[field] || Boolean(errorMessages[field]),
    ),
    [errorMessages],
  );

  const doSignup = useCallback(async () => {
    try {
      await signup(fields);
      setIsSubmit(false);
      setIsSuccess(true);
    } catch (error) {
      setIsSubmit(false);
      let errorMessage = translate('ERROR.NEWOR_INTERNAL_SERVER_ERROR');
      const errorCode = error?.response?.data?.code;
      if (errorCode) {
        errorMessage = translate(`ERROR.${errorCode}`);
      }
      toast.show({
        render: () => <ToastAlert status="error" message={errorMessage} />,
        placement: 'top',
      });
    }
  }, [fields]);

  useEffect(() => {
    if (isSubmit && !isFormError) {
      doSignup();
    } else {
      setIsSubmit(false);
    }
  }, [navigation, isSubmit, isFormError]);

  return (
    <Flex flex={1}>
      <Flex flex={1} />
      <Flex flex={4} direction="row" p="5" justifyContent="center">
        <FormContainer>
          <TitleBar color={theme.color.SECONDARY_100}>
            <Translation tkey="WELCOME_TO_NEWOR" as={Title} />
          </TitleBar>
          <Center><Image alt={CONSTANT.APP_NAME} source={NEWOR} size="xl" /></Center>
          <For each="field" index="index" of={FIELDS}>
            <FormControl key={`signup-field-${index}`} isInvalid={errorMessages[field.key]}>
              <InputField
                testID={`signup-input-${field.key}`}
                placeholder={translate(field.placeholder)}
                value={fields[field.key]}
                onChangeText={(value) => handleFieldChange(field.key, value)}
                onBlur={() => handleValidation()}
                type={
                  field.key === CONSTANT.AUTH_LITERAL.PASSWORD && CONSTANT.AUTH_LITERAL.PASSWORD
                }
                isInvalid={errorMessages[field.key]}
              />
              <FormControl.ErrorMessage>
                {errorMessages[field.key]}
              </FormControl.ErrorMessage>
            </FormControl>
          </For>
          <Translation
            tkey="SIGNUP"
            as={AppButton}
            testID="signup-submit"
            onPress={handleSubmit}
            isDisabled={isFormError}
            isLoading={isSubmit}
            variant="primary"
            mb={5}
          />
          <Flex direction="row" justifyContent="center">
            <Translation
              tkey="ALREADY_HAVE_ACCOUNT"
              as={LoginLink}
            />
            &nbsp;
            <Translation
              tkey="LOGIN"
              as={LoginLink}
              link
              onPress={() => navigation.navigate(ROUTE.LOGIN)}
            />
          </Flex>
        </FormContainer>
      </Flex>
      <Flex flex={1} />
      <AlertDialog testID="signup-success-modal" avoidKeyboard size="xl" isOpen={isSuccess} onClose={() => setIsSuccess(false)}>
        <AlertDialog.Content>
          <AlertDialog.CloseButton />
          <Translation tkey="SIGNUP_SUCCESS" as={AlertDialog.Header} />
          <AlertDialog.Body>
            <Center>
              <Image alt={translate('SIGNUP_SUCCESS')} source={NEWOR_SUCCESS} size="2xl" />
              <Translation
                tkey="PLEASE_VERIFY_EMAIL"
                as={LoginLink}
              />
            </Center>
          </AlertDialog.Body>
          <AlertDialog.Footer>
            <Translation
              tkey="LOGIN"
              as={AppButton}
              variant="primary"
              flex={1}
              testID="signup-success-login"
              onPress={() => {
                setIsSuccess(false);
                navigation.navigate(ROUTE.LOGIN);
              }}
            />
          </AlertDialog.Footer>
        </AlertDialog.Content>
      </AlertDialog>
    </Flex>
  );
};

export default withBackground({ Component: Signup, Background: Background2 });
