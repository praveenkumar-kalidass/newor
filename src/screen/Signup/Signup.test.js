import React from 'react';

import { signup } from '../../api/user';
import { act, render, fireEvent } from '../../test/util';
import Signup from './Signup';

const mockNavigation = {
  navigate: jest.fn(),
};
jest.mock('@react-navigation/native', () => {
  const reactNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...reactNavigation,
    useNavigation: () => mockNavigation,
  };
});
const mockToast = { show: jest.fn() };
jest.mock('native-base', () => {
  const nativeBase = jest.requireActual('native-base');
  return {
    ...nativeBase,
    useToast: () => mockToast,
  };
});
jest.mock('../../api/user', () => ({
  signup: jest.fn(),
}));

describe('Signup', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', async () => {
    const container = await render(<Signup />);

    expect(container).toMatchSnapshot();
  });

  it('should show toast when signup fails', async () => {
    signup.mockRejectedValueOnce({ response: { data: { code: 'NEWOR_INTERNAL_SERVER_ERROR' } } });

    const { getByTestId } = await render(<Signup />);

    await act(async () => {
      await fireEvent.changeText(getByTestId('signup-input-firstName'), 'Test');
      await fireEvent(getByTestId('signup-input-firstName'), 'onBlur');
      await fireEvent.changeText(getByTestId('signup-input-lastName'), 'Newor');
      await fireEvent(getByTestId('signup-input-lastName'), 'onBlur');
      await fireEvent.changeText(getByTestId('signup-input-email'), 'test@newor.com');
      await fireEvent(getByTestId('signup-input-email'), 'onBlur');
      await fireEvent.changeText(getByTestId('signup-input-password'), '123456');
      await fireEvent(getByTestId('signup-input-password'), 'onBlur');
    });

    await act(async () => {
      await fireEvent.press(getByTestId('signup-submit'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(1);
  });

  it('should show toast and navigate when signup is success', async () => {
    signup.mockResolvedValueOnce();

    const { getByTestId } = await render(<Signup />);

    await act(async () => {
      await fireEvent.changeText(getByTestId('signup-input-firstName'), 'Test');
      await fireEvent(getByTestId('signup-input-firstName'), 'onBlur');
      await fireEvent.changeText(getByTestId('signup-input-lastName'), 'Newor');
      await fireEvent(getByTestId('signup-input-lastName'), 'onBlur');
      await fireEvent.changeText(getByTestId('signup-input-email'), 'test@newor.com');
      await fireEvent(getByTestId('signup-input-email'), 'onBlur');
      await fireEvent.changeText(getByTestId('signup-input-password'), '123456');
      await fireEvent(getByTestId('signup-input-password'), 'onBlur');
    });

    await act(async () => {
      await fireEvent.press(getByTestId('signup-submit'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(1);

    await act(() => {
      mockToast.show.mock.calls[0][0].onCloseComplete();
    });

    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith('LOGIN');
  });
});
