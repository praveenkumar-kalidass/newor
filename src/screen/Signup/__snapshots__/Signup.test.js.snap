// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    />
    <View
      dataSet={Object {}}
      style={
        Object {
          "display": "flex",
          "flex": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 20,
              "flex": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
              },
              Array [
                Object {},
              ],
            ]
          }
          styleProps={Object {}}
        >
          <View
            dataSet={Object {}}
            style={
              Array [
                Object {
                  "backgroundColor": "#00238a",
                  "color": "#00238a",
                  "marginRight": 8,
                  "paddingRight": 8,
                },
                Array [
                  Object {},
                ],
              ]
            }
          />
          <Text
            dataSet={Object {}}
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "letterSpacing": 0,
                  "lineHeight": 30,
                  "textDecorationLine": undefined,
                },
                Array [
                  Object {
                    "textTransform": "capitalize",
                  },
                ],
              ]
            }
          >
            Welcome to Newor!
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <Image
            accessibilityLabel="newor"
            alt="newor"
            dataSet={Object {}}
            onError={[Function]}
            source={
              Object {
                "testUri": "../../../src/asset/newor.png",
              }
            }
            style={
              Object {
                "height": 128,
                "maxWidth": "100%",
                "width": 128,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          isInvalid=""
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 20,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessibilityDescribedBy="field-react-aria-1-feedback"
              accessible={true}
              dataSet={Object {}}
              disabled={false}
              editable={true}
              isInvalid=""
              isRequired={false}
              nativeID="field-react-aria-1"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="First name"
              placeholderTextColor="#a3a3a3"
              readOnly={false}
              required={false}
              secureTextEntry={false}
              selectionColor="#1f2937"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                  ],
                ]
              }
              testID="signup-input-firstName"
              value=""
            />
          </View>
        </View>
        <View
          dataSet={Object {}}
          isInvalid=""
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 20,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessibilityDescribedBy="field-react-aria-2-feedback"
              accessible={true}
              dataSet={Object {}}
              disabled={false}
              editable={true}
              isInvalid=""
              isRequired={false}
              nativeID="field-react-aria-2"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Last name"
              placeholderTextColor="#a3a3a3"
              readOnly={false}
              required={false}
              secureTextEntry={false}
              selectionColor="#1f2937"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                  ],
                ]
              }
              testID="signup-input-lastName"
              value=""
            />
          </View>
        </View>
        <View
          dataSet={Object {}}
          isInvalid=""
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 20,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessibilityDescribedBy="field-react-aria-3-feedback"
              accessible={true}
              dataSet={Object {}}
              disabled={false}
              editable={true}
              isInvalid=""
              isRequired={false}
              nativeID="field-react-aria-3"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Email"
              placeholderTextColor="#a3a3a3"
              readOnly={false}
              required={false}
              secureTextEntry={false}
              selectionColor="#1f2937"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                  ],
                ]
              }
              testID="signup-input-email"
              value=""
            />
          </View>
        </View>
        <View
          dataSet={Object {}}
          isInvalid=""
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 20,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessibilityDescribedBy="field-react-aria-4-feedback"
              accessible={true}
              dataSet={Object {}}
              disabled={false}
              editable={true}
              isInvalid=""
              isRequired={false}
              nativeID="field-react-aria-4"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Password"
              placeholderTextColor="#a3a3a3"
              readOnly={false}
              required={false}
              secureTextEntry={true}
              selectionColor="#1f2937"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                  ],
                ]
              }
              testID="signup-input-password"
              value=""
            />
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ad0000",
                "borderRadius": 4,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 20,
                "opacity": 0.4,
                "paddingBottom": 10,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 10,
              },
              Array [
                Object {},
              ],
            ]
          }
          testID="signup-submit"
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={Object {}}
              style={Object {}}
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#fafafa",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Signup
              </Text>
            </View>
          </View>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 24,
                  "textDecorationLine": undefined,
                },
                Array [
                  Object {},
                ],
              ]
            }
          >
            Already have an account?
             
          </Text>
          <Text
            dataSet={Object {}}
            link={true}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "color": "#3498db",
                  "fontFamily": undefined,
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 24,
                  "textDecorationLine": "underline",
                },
                Array [
                  Object {},
                ],
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
